src/core/src/op/group_normalization.cpp

```diff
@@ -0,0 +1,66 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "openvino/op/group_normalization.hpp"
+
+#include "itt.hpp"
+#include "ngraph/validation_util.hpp"
+#include "openvino/core/attribute_visitor.hpp"
+
+using namespace ov;
+
+op::v12::GroupNormalization::GroupNormalization(const Output<Node>& data,
+                                                const Output<Node>& scale,
+                                                const Output<Node>& bias,
+                                                int64_t num_groups,
+                                                double epsilon)
+    : Op({data, scale, bias}),
+      m_num_groups{num_groups},
+      m_epsilon{epsilon} {
+    constructor_validate_and_infer_types();
+}
+
+bool op::v12::GroupNormalization::visit_attributes(AttributeVisitor& visitor) {
+    OV_OP_SCOPE(v12_GroupNormalization_visit_attributes);
+    visitor.on_attribute("num_groups", m_num_groups);
+    visitor.on_attribute("epsilon", m_epsilon);
+    return true;
+}
+
+void op::v12::GroupNormalization::validate_and_infer_types() {
```

Comment:
Consider moving the shape validation to the common shape_infer function in [shape_inference](https://github.com/openvinotoolkit/openvino/tree/master/src/core/shape_inference/include).

Reply:
Just for information what should be check regarding shape inference.
1. Dimension and labels propagation (do check on dynamic rank input, static rank and interval dimension with labels)
2. Add template implementation for shape inference in future it can be re used with CPU or GPU plugin and check if possible to call `shape_infer` function where operator created with default ctor (best option no set inputs to simylate GPU usage)
These point will not apply to this op as it just copy input shape to output.
3. If output shape depends on input values then there should be check if this input (e.g as result of ShapeOf) which can have interval values and labels are propageted correctly to output.
4. If operator is used in SsapeOf sub-graph then it should have evaluate upper/lower and labels to propagate the values and labels.

Reply:
I confirm, also points 3 and 4 seem to be not applicable for GroupNormalization.